artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/LinkedListImpl.java

```diff
@@ -102,6 +103,17 @@ public void addHead(E e) {
       size++;
    }
 
+   @Override
+   public E get(int position) {
+      Node<E> current = head.next;
+
+      for (int i = 0; i < position && current != null; i++) {
+         current = current.next;
+      }
+
+      return current.val();
```

Comment:
This assumes current exists but it may not (which the for handles). Should throw index out of bounds rather than NPE?